docs_src/microservices/security/Ch-SSH-Tunneling-HowToSecureDeviceServices.md

```diff
@@ -0,0 +1,297 @@
+# Security for EdgeX Stack
+
+This page describes one of options to secure the EdgeX software stack with running remote device services like device-virtual, device-rest, device-mqtt, and so on, via secure two-way SSH-tunnelings.
+
+## Basic SSH-Tunneling
+
+In this option to secure the EdgeX software stack, SSH tunneling is utilized. The basic idea is to create a secure SSH connection between  a local machine, the primary host, and a remote machine, the secondary, in which some micro-services or applications can be relayed.   In this particular example, the primary host is  running the whole EdgeX core services including core services and security services but **without** any device service.  The device services are running in the secondary or the remote machine.  
+
+The communication is secure because SSH port forwarding connection is encrypted by default.
+
+The SSH communication is established by introducing some extra SSH-related services:
+
+1) device-ssh-proxy: this is the service with ssh client opening up the SSH communication between the primary and the secondary
+
+2) device-ssh-remote: this is actually the SSH server or daemon service together with device services running on the remote machine
+
+The high-level diagram is shown as follows:
+
+![image](ssh-tunneling_device.png)
+
+"Top level diagram for SSH tunneling for device services"
+
+In the primary, the SSH tunneling initiated by device-ssh-proxy service to the remote running device services.  The dependencies that remote device services needed are reversely tunneling from the primary machine.
+
+## How to- Reference implementation example
+
+The whole reference implementation example can be found in this repository: 
+[HoldingRepoSshTunneling](https://github.com/edgexfoundry-holding/external-device-security-examples/ssh-tunneling/README.md)
+
+### Setup remote running Virtual Machine
+
+In the example setup, `vagrant` is used on the top of `Virtual Box` to set up as the secondary/remote VM. The network port for ssh standard port 22 is mapped into 2222 for `vagrant ssh` itself and the forwarded port is also mapped on the VM network for port 2223 to the host machine port 2223.  This port 2223 is used for the ssh daemon Docker container that will be introduced later on below.
+
+Once you have downloaded the vagrant from Hashicorp website, typical vagrant setup for the first time can be done via command `./vagrant init` and it will generate the Vagrant configuration file.
+
+Here is the Vagrant file used to create the remote machine:
+
+[remote VM Vagrant file with docker and docker-compose installed](https://github.com/edgexfoundry-holding/external-device-security-examples/ssh-tunneling/Vagrantfile)
+
+### SSH Tunneling: Setup the SSH server on the remote machine
+
+Use the example from Docker hub, an example of ssh damenon or ssh server can be setup pretty easily as a Docker container: https://docs.docker.com/engine/examples/running_ssh_service/
```

Comment:
don't use example twice in the same sentence. Additional, stay away from terms like easily since it may be easy for you but not for someone else.

Try: "For an example of how to run an ssh server, checkout https://docs.docker.com/engine/examples/running_ssh_service/ for instructions.